services/httpd/handler_test.go

```diff
@@ -698,6 +698,60 @@ func TestHandler_XForwardedFor(t *testing.T) {
 	}
 }
 
+func TestHandler_XRequestID(t *testing.T) {
+	var buf bytes.Buffer
+	h := NewHandler(false)
+	h.CLFLogger = log.New(&buf, "", 0)
+
+	cases := []struct {
+		XRequestID string
+		RequestId  string
+	}{
+		{XRequestID: "abc123", RequestId: ""},          // X-Request-ID is used.
+		{XRequestID: "", RequestId: "foobarzoo"},       // Request-Id is used.
+		{XRequestID: "abc123", RequestId: "foobarzoo"}, // X-Request-ID takes precedence.
+		{XRequestID: "", RequestId: ""},                // v1 UUID generated.
+	}
+
+	for _, c := range cases {
+		t.Run(fmt.Sprintf("X-Request-ID: %q Request-Id: %q", c.XRequestID, c.RequestId), func(t *testing.T) {
+			buf.Reset() // Reset the buffer.
+			req := MustNewRequest("GET", "/ping", nil)
+			req.RemoteAddr = "127.0.0.1"
+
+			// Set the relevant request ID headers
+			req.Header.Set("X-Request-ID", c.XRequestID)
+			req.Header.Set("Request-Id", c.RequestId)
+
+			w := httptest.NewRecorder()
+			h.ServeHTTP(w, req)
+
+			parts := strings.Split(buf.String(), " ")
```

Comment:
There probably is but it would be convoluted I expect, as we would have to parse a log entry. I could add a comment explaining that this will need to be adjusted if we change what we output in the HTTP logs